### ISSUE TITLE
shapely 2.01 parallel_offset different behaviour from 1.8.4

### ISSUE THREAD

**Issue by xmarduel** [shapely/shapely#1789]
With a "complex linestring", 1.8.4 parallel_offset used to return a MultiLineString composed in my example of 3 distinct Linestrings, each of them completely separated from each other
With shapely 2.0.1, I only obtain 1 single LineString, only the first one given in 1.8.4. The two others are missing.

Windows 2.0.1 installed from PyPI using pip
![offset-input-linestring](https://user-images.githubusercontent.com/28778239/225379901-2c078730-6842-4899-806b-99be333181b1.png)
![offset-3-linestrings](https://user-images.githubusercontent.com/28778239/225379918-0a1d454e-c3be-4c62-9070-1dc71b2272df.png)

**Comment by jorisvandenbossche** [comment]
Similarly as in the other issue, this is probably due to a change in the implementation in GEOS 3.11. See https://github.com/libgeos/geos/issues/816 for some discussion.

Do you have a reproducible example? Then I would be happy to test it with the latest GEOS main.

**Comment by xmarduel** [comment]
Yes sure
[wkt.txt](https://github.com/shapely/shapely/files/11010797/wkt.txt)

```python
from shapely import wkt
import shapely

import matplotlib
import matplotlib.pyplot as plt


def plot(linestring):
    plt.figure()

    x = linestring.coords.xy[0]
    y = linestring.coords.xy[1]
    plt.plot(x,y)
    plt.show()

fp = open("wkt.txt", "r")
ww = fp.read()
fp.close()

ww

linestring = shapely.wkt.loads(ww)
linestring

plot(linestring)

offset = linestring.parallel_offset(0.25, 'left', resolution=16)
offset

print( "shapely version ", shapely.__version__, "offset type :", offset.geom_type)

# shapely 1.8.4 -> MultiLineString, 3 lines
if shapely.__version__ == '1.8.4':
    print(" shapely 1.8.4: nb lines = ", len(list(offset.geoms)))

    for line in offset.geoms:
        plot(line)

# shapely 2.0.1 -> LineString, 1 line, 2 others missing
if shapely.__version__ == '2.0.1':
    plot(offset)

```

Many thanks.
XM

**Comment by jorisvandenbossche** [comment]
Yes, I can confirm that this seems fixed with GEOS main (https://github.com/libgeos/geos/issues/816).

Plotting the original line and the offset for shapely 1.8 using GEOS 3.10.1, shapely 2.0 using GEOS 3.11.1 and shapely 2.0 using GEOS 3.12.dev main:

GEOS 3.10:

![offset_shapely18_GEOS_310](https://user-images.githubusercontent.com/1020496/226177338-30996943-ef8d-4359-9084-92eea26f49d4.png)

GEOS 3.11:

![offset_shapely20_GEOS_311](https://user-images.githubusercontent.com/1020496/226177343-6982f8c3-f280-4e4e-8315-d9a9d1dc9cdc.png)

GEOS 3.12:

![offset_GEOS_main](https://user-images.githubusercontent.com/1020496/226177356-518f85e3-1070-404e-bd80-71d6e079671e.png)


<details>

```python
import shapely

with open("../Downloads/wkt.txt", "r") as fp:
    linestring = shapely.from_wkt(fp.read())

offset = linestring.parallel_offset(0.25, 'left', resolution=16)

from shapely.plotting import plot_line
plot_line(linestring)
plot_line(offset, color="C2")
```

</details>
